src/nbl/video/utilities/IUtilities.cpp

```diff
@@ -1,171 +1,75 @@
+        for (uint32_t i = regionIterator.getCurrentRegion(); i < regions.size(); ++i)
+        {
             const asset::IImage::SBufferCopy & region = regions[i];
-
-            auto subresourceSize = dstImage->getMipSize(region.imageSubresource.mipLevel);
-
-            assert(static_cast<uint32_t>(region.imageSubresource.aspectMask) != 0u);
-            assert(core::isPoT(static_cast<uint32_t>(region.imageSubresource.aspectMask)) && "region.aspectMask should only have a single bit set.");
-            // Validate Region
-            // canTransferMipLevelsPartially
-            if(!canTransferMipLevelsPartially)
-            {
-                assert(region.imageOffset.x == 0 && region.imageOffset.y == 0 && region.imageOffset.z == 0);
-                assert(region.imageExtent.width == subresourceSize.x && region.imageExtent.height == subresourceSize.y && region.imageExtent.depth == subresourceSize.z);
-            }
-
-            // region.imageOffset.{xyz} should be multiple of minImageTransferGranularity.{xyz} scaled up by block size
-            bool isImageOffsetAlignmentValid =
-                (region.imageOffset.x % (minImageTransferGranularity.width  * texelBlockDim.x) == 0) &&
-                (region.imageOffset.y % (minImageTransferGranularity.height * texelBlockDim.y) == 0) &&
-                (region.imageOffset.z % (minImageTransferGranularity.depth  * texelBlockDim.z) == 0);
-            assert(isImageOffsetAlignmentValid);
-
-            // region.imageExtent.{xyz} should be multiple of minImageTransferGranularity.{xyz} scaled up by block size,
-            // OR ELSE (region.imageOffset.{x/y/z} + region.imageExtent.{width/height/depth}) MUST be equal to subresource{Width,Height,Depth}
-            bool isImageExtentAlignmentValid = 
-                (region.imageExtent.width  % (minImageTransferGranularity.width  * texelBlockDim.x) == 0 || (region.imageOffset.x + region.imageExtent.width   == subresourceSize.x)) && 
-                (region.imageExtent.height % (minImageTransferGranularity.height * texelBlockDim.y) == 0 || (region.imageOffset.y + region.imageExtent.height  == subresourceSize.y)) &&
-                (region.imageExtent.depth  % (minImageTransferGranularity.depth  * texelBlockDim.z) == 0 || (region.imageOffset.z + region.imageExtent.depth   == subresourceSize.z));
-            assert(isImageExtentAlignmentValid);
-
-            bool isImageExtentAndOffsetValid = 
-                (region.imageExtent.width + region.imageOffset.x <= subresourceSize.x) &&
-                (region.imageExtent.height + region.imageOffset.y <= subresourceSize.y) &&
-                (region.imageExtent.depth + region.imageOffset.z <= subresourceSize.z);
-            assert(isImageExtentAndOffsetValid);
-
             auto imageExtent = core::vector3du32_SIMD(region.imageExtent.width, region.imageExtent.height, region.imageExtent.depth);
             auto imageExtentInBlocks = texelBlockInfo.convertTexelsToBlocks(imageExtent);
```

Comment:
careful about `convertTexelsToBlocks` rounding mode, when transfering _whole_ mips of compressed textures you kinda need to round up

Reply:
```c		
inline auto convertTexelsToBlocks(const core::vector3du32_SIMD& coord) const
{
	return (coord+maxBlockCoord)/dimension;
}
```